### ISSUE TITLE
Build fail on UbuntuDDE 20.04 Remix

### ISSUE THREAD

**Issue by zeletrik** [SeptemberHX/dde-top-panel#28]
```
% cmake ..
-- The C compiler identification is GNU 9.3.0
-- The CXX compiler identification is GNU 9.3.0
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
bash: translate_generation.sh: No such file or directory
-- Found PkgConfig: /usr/bin/pkg-config (found version "0.29.1") 
OS: ubuntu, Version: 20.04
Deepin OS Type: Focal
Deepin OS Version: 20.04 LTS
-- Checking for modules 'xcb-ewmh;x11'
--   Found xcb-ewmh, version 0.4.1
--   Found x11, version 1.6.9
-- Checking for module 'dframeworkdbus'
--   Found dframeworkdbus, version 2.0
-- Checking for module 'gsettings-qt'
--   Found gsettings-qt, version 1.0.0
-- Checking for module 'dtkgui'
--   Found dtkgui, version 2.1.1
-- Configuring done
-- Generating done
-- Build files have been written to: /home/zeletrik/Downloads/0.5.5/dde-top-panel-0.5.5/build

% make

Scanning dependencies of target dde-top-panel_autogen
[  1%] Automatic MOC and UIC for target dde-top-panel
AutoMoc warning
---------------
"SRC:/appmenu/dbusmenu_interface.cpp"
includes the moc file "dbusmenu_interface.moc", but does not contain a Q_OBJECT, Q_GADGET or Q_NAMESPACE macro.

AutoMoc warning
---------------
"SRC:/appmenu/menuimporteradaptor.cpp"
includes the moc file "menuimporteradaptor.moc", but does not contain a Q_OBJECT, Q_GADGET or Q_NAMESPACE macro.

AutoMoc: /home/zeletrik/Downloads/0.5.5/dde-top-panel-0.5.5/appmenu/dbusmenu_interface.cpp:0: Note: No relevant classes found. No output generated.
AutoMoc: /home/zeletrik/Downloads/0.5.5/dde-top-panel-0.5.5/appmenu/menuimporteradaptor.cpp:0: Note: No relevant classes found. No output generated.
[  1%] Built target dde-top-panel_autogen
[  3%] Automatic RCC for frame.qrc
[  5%] Automatic RCC for item/item.qrc
Scanning dependencies of target dde-top-panel
[  7%] Building CXX object frame/CMakeFiles/dde-top-panel.dir/dde-top-panel_autogen/mocs_compilation.cpp.o
In file included from /home/zeletrik/Downloads/0.5.5/dde-top-panel-0.5.5/frame/item/dockitem.h:26,
                 from /home/zeletrik/Downloads/0.5.5/dde-top-panel-0.5.5/frame/item/pluginsitem.h:25,
                 from /home/zeletrik/Downloads/0.5.5/dde-top-panel-0.5.5/frame/controller/dockpluginscontroller.h:25,
                 from /home/zeletrik/Downloads/0.5.5/dde-top-panel-0.5.5/frame/controller/dockitemmanager.h:25,
                 from /home/zeletrik/Downloads/0.5.5/dde-top-panel-0.5.5/build/frame/dde-top-panel_autogen/NPUKSKGWT2/moc_dockitemmanager.cpp:9,
                 from /home/zeletrik/Downloads/0.5.5/dde-top-panel-0.5.5/build/frame/dde-top-panel_autogen/mocs_compilation.cpp:6:
/home/zeletrik/Downloads/0.5.5/dde-top-panel-0.5.5/frame/util/dockpopupwindow.h:30:1: error: ‘DGUI_USE_NAMESPACE’ does not name a type; did you mean ‘DTK_USE_NAMESPACE’?
   30 | DGUI_USE_NAMESPACE
      | ^~~~~~~~~~~~~~~~~~
      | DTK_USE_NAMESPACE
In file included from /home/zeletrik/Downloads/0.5.5/dde-top-panel-0.5.5/frame/item/pluginsitem.h:25,
                 from /home/zeletrik/Downloads/0.5.5/dde-top-panel-0.5.5/frame/controller/dockpluginscontroller.h:25,
                 from /home/zeletrik/Downloads/0.5.5/dde-top-panel-0.5.5/frame/controller/dockitemmanager.h:25,
                 from /home/zeletrik/Downloads/0.5.5/dde-top-panel-0.5.5/build/frame/dde-top-panel_autogen/NPUKSKGWT2/moc_dockitemmanager.cpp:9,
                 from /home/zeletrik/Downloads/0.5.5/dde-top-panel-0.5.5/build/frame/dde-top-panel_autogen/mocs_compilation.cpp:6:
/home/zeletrik/Downloads/0.5.5/dde-top-panel-0.5.5/frame/item/dockitem.h:120:21: error: ‘DockPopupWindow’ was not declared in this scope; did you mean ‘showPopupWindow’?
  120 |     static QPointer<DockPopupWindow> PopupWindow;
      |                     ^~~~~~~~~~~~~~~
      |                     showPopupWindow
/home/zeletrik/Downloads/0.5.5/dde-top-panel-0.5.5/frame/item/dockitem.h:120:36: error: template argument 1 is invalid
  120 |     static QPointer<DockPopupWindow> PopupWindow;
      |                                    ^
In file included from /home/zeletrik/Downloads/0.5.5/dde-top-panel-0.5.5/build/frame/dde-top-panel_autogen/mocs_compilation.cpp:22:
/home/zeletrik/Downloads/0.5.5/dde-top-panel-0.5.5/build/frame/dde-top-panel_autogen/KH43KSYMFX/moc_dockpopupwindow.cpp:100:6: error: ‘DockPopupWindow’ has not been declared
  100 | void DockPopupWindow::qt_static_metacall(QObject *_o, QMetaObject::Call _c, int _id, void **_a)
      |      ^~~~~~~~~~~~~~~
/home/zeletrik/Downloads/0.5.5/dde-top-panel-0.5.5/build/frame/dde-top-panel_autogen/KH43KSYMFX/moc_dockpopupwindow.cpp: In function ‘void qt_static_metacall(QObject*, QMetaObject::Call, int, void**)’:
/home/zeletrik/Downloads/0.5.5/dde-top-panel-0.5.5/build/frame/dde-top-panel_autogen/KH43KSYMFX/moc_dockpopupwindow.cpp:103:32: error: ‘DockPopupWindow’ does not name a type
  103 |         auto *_t = static_cast<DockPopupWindow *>(_o);
      |                                ^~~~~~~~~~~~~~~
/home/zeletrik/Downloads/0.5.5/dde-top-panel-0.5.5/build/frame/dde-top-panel_autogen/KH43KSYMFX/moc_dockpopupwindow.cpp:103:48: error: expected ‘>’ before ‘*’ token
  103 |         auto *_t = static_cast<DockPopupWindow *>(_o);
      |                                                ^
/home/zeletrik/Downloads/0.5.5/dde-top-panel-0.5.5/build/frame/dde-top-panel_autogen/KH43KSYMFX/moc_dockpopupwindow.cpp:103:48: error: expected ‘(’ before ‘*’ token
  103 |         auto *_t = static_cast<DockPopupWindow *>(_o);
      |                                                ^
      |                                                (
/home/zeletrik/Downloads/0.5.5/dde-top-panel-0.5.5/build/frame/dde-top-panel_autogen/KH43KSYMFX/moc_dockpopupwindow.cpp:103:49: error: expected primary-expression before ‘>’ token
  103 |         auto *_t = static_cast<DockPopupWindow *>(_o);
      |                                                 ^
/home/zeletrik/Downloads/0.5.5/dde-top-panel-0.5.5/build/frame/dde-top-panel_autogen/KH43KSYMFX/moc_dockpopupwindow.cpp:103:54: error: expected ‘)’ before ‘;’ token
  103 |         auto *_t = static_cast<DockPopupWindow *>(_o);
      |                                                      ^
      |                                                      )
In file included from /home/zeletrik/Downloads/0.5.5/dde-top-panel-0.5.5/build/frame/dde-top-panel_autogen/mocs_compilation.cpp:22:
/home/zeletrik/Downloads/0.5.5/dde-top-panel-0.5.5/build/frame/dde-top-panel_autogen/KH43KSYMFX/moc_dockpopupwindow.cpp:120:28: error: expected ‘;’ before ‘(’ token
  120 |             using _t = void (DockPopupWindow::*)() const;
      |                            ^~
      |                            ;
/home/zeletrik/Downloads/0.5.5/dde-top-panel-0.5.5/build/frame/dde-top-panel_autogen/KH43KSYMFX/moc_dockpopupwindow.cpp:121:35: error: ‘_t’ does not name a type; did you mean ‘_o’?
  121 |             if (*reinterpret_cast<_t *>(_a[1]) == static_cast<_t>(&DockPopupWindow::accept)) {
      |                                   ^~
      |                                   _o
/home/zeletrik/Downloads/0.5.5/dde-top-panel-0.5.5/build/frame/dde-top-panel_autogen/KH43KSYMFX/moc_dockpopupwindow.cpp:121:38: error: expected ‘>’ before ‘*’ token
  121 |             if (*reinterpret_cast<_t *>(_a[1]) == static_cast<_t>(&DockPopupWindow::accept)) {
      |                                      ^
/home/zeletrik/Downloads/0.5.5/dde-top-panel-0.5.5/build/frame/dde-top-panel_autogen/KH43KSYMFX/moc_dockpopupwindow.cpp:121:38: error: expected ‘(’ before ‘*’ token
  121 |             if (*reinterpret_cast<_t *>(_a[1]) == static_cast<_t>(&DockPopupWindow::accept)) {
      |                                      ^
      |                                      (
/home/zeletrik/Downloads/0.5.5/dde-top-panel-0.5.5/build/frame/dde-top-panel_autogen/KH43KSYMFX/moc_dockpopupwindow.cpp:121:39: error: expected primary-expression before ‘>’ token
  121 |             if (*reinterpret_cast<_t *>(_a[1]) == static_cast<_t>(&DockPopupWindow::accept)) {
      |                                       ^
/home/zeletrik/Downloads/0.5.5/dde-top-panel-0.5.5/build/frame/dde-top-panel_autogen/KH43KSYMFX/moc_dockpopupwindow.cpp:121:63: error: ‘_t’ does not name a type; did you mean ‘_o’?
  121 |             if (*reinterpret_cast<_t *>(_a[1]) == static_cast<_t>(&DockPopupWindow::accept)) {
      |                                                               ^~
      |                                                               _o
/home/zeletrik/Downloads/0.5.5/dde-top-panel-0.5.5/build/frame/dde-top-panel_autogen/KH43KSYMFX/moc_dockpopupwindow.cpp:121:68: error: ‘DockPopupWindow’ has not been declared
  121 |             if (*reinterpret_cast<_t *>(_a[1]) == static_cast<_t>(&DockPopupWindow::accept)) {
      |                                                                    ^~~~~~~~~~~~~~~
/home/zeletrik/Downloads/0.5.5/dde-top-panel-0.5.5/build/frame/dde-top-panel_autogen/KH43KSYMFX/moc_dockpopupwindow.cpp:121:93: error: expected ‘)’ before ‘{’ token
  121 |             if (*reinterpret_cast<_t *>(_a[1]) == static_cast<_t>(&DockPopupWindow::accept)) {
      |                ~                                                                            ^~
      |                                                                                             )
/home/zeletrik/Downloads/0.5.5/dde-top-panel-0.5.5/build/frame/dde-top-panel_autogen/KH43KSYMFX/moc_dockpopupwindow.cpp:125:9: error: expected primary-expression before ‘}’ token
  125 |         }
      |         ^
/home/zeletrik/Downloads/0.5.5/dde-top-panel-0.5.5/build/frame/dde-top-panel_autogen/KH43KSYMFX/moc_dockpopupwindow.cpp:127:28: error: expected ‘;’ before ‘(’ token
  127 |             using _t = void (DockPopupWindow::*)();
      |                            ^~
      |                            ;
/home/zeletrik/Downloads/0.5.5/dde-top-panel-0.5.5/build/frame/dde-top-panel_autogen/KH43KSYMFX/moc_dockpopupwindow.cpp:128:35: error: ‘_t’ does not name a type; did you mean ‘_o’?
  128 |             if (*reinterpret_cast<_t *>(_a[1]) == static_cast<_t>(&DockPopupWindow::unusedSignal)) {
      |                                   ^~
      |                                   _o
/home/zeletrik/Downloads/0.5.5/dde-top-panel-0.5.5/build/frame/dde-top-panel_autogen/KH43KSYMFX/moc_dockpopupwindow.cpp:128:38: error: expected ‘>’ before ‘*’ token
  128 |             if (*reinterpret_cast<_t *>(_a[1]) == static_cast<_t>(&DockPopupWindow::unusedSignal)) {
      |                                      ^
/home/zeletrik/Downloads/0.5.5/dde-top-panel-0.5.5/build/frame/dde-top-panel_autogen/KH43KSYMFX/moc_dockpopupwindow.cpp:128:38: error: expected ‘(’ before ‘*’ token
  128 |             if (*reinterpret_cast<_t *>(_a[1]) == static_cast<_t>(&DockPopupWindow::unusedSignal)) {
      |                                      ^
      |                                      (
/home/zeletrik/Downloads/0.5.5/dde-top-panel-0.5.5/build/frame/dde-top-panel_autogen/KH43KSYMFX/moc_dockpopupwindow.cpp:128:39: error: expected primary-expression before ‘>’ token
  128 |             if (*reinterpret_cast<_t *>(_a[1]) == static_cast<_t>(&DockPopupWindow::unusedSignal)) {
      |                                       ^
/home/zeletrik/Downloads/0.5.5/dde-top-panel-0.5.5/build/frame/dde-top-panel_autogen/KH43KSYMFX/moc_dockpopupwindow.cpp:128:63: error: ‘_t’ does not name a type; did you mean ‘_o’?
  128 |             if (*reinterpret_cast<_t *>(_a[1]) == static_cast<_t>(&DockPopupWindow::unusedSignal)) {
      |                                                               ^~
      |                                                               _o
/home/zeletrik/Downloads/0.5.5/dde-top-panel-0.5.5/build/frame/dde-top-panel_autogen/KH43KSYMFX/moc_dockpopupwindow.cpp:128:68: error: ‘DockPopupWindow’ has not been declared
  128 |             if (*reinterpret_cast<_t *>(_a[1]) == static_cast<_t>(&DockPopupWindow::unusedSignal)) {
      |                                                                    ^~~~~~~~~~~~~~~
/home/zeletrik/Downloads/0.5.5/dde-top-panel-0.5.5/build/frame/dde-top-panel_autogen/KH43KSYMFX/moc_dockpopupwindow.cpp:128:99: error: expected ‘)’ before ‘{’ token
  128 |             if (*reinterpret_cast<_t *>(_a[1]) == static_cast<_t>(&DockPopupWindow::unusedSignal)) {
      |                ~                                                                                  ^~
      |                                                                                                   )
/home/zeletrik/Downloads/0.5.5/dde-top-panel-0.5.5/build/frame/dde-top-panel_autogen/KH43KSYMFX/moc_dockpopupwindow.cpp:132:9: error: expected primary-expression before ‘}’ token
  132 |         }
      |         ^
/home/zeletrik/Downloads/0.5.5/dde-top-panel-0.5.5/build/frame/dde-top-panel_autogen/KH43KSYMFX/moc_dockpopupwindow.cpp:118:14: warning: unused variable ‘result’ [-Wunused-variable]
  118 |         int *result = reinterpret_cast<int *>(_a[0]);
      |              ^~~~~~
/home/zeletrik/Downloads/0.5.5/dde-top-panel-0.5.5/build/frame/dde-top-panel_autogen/KH43KSYMFX/moc_dockpopupwindow.cpp: At global scope:
/home/zeletrik/Downloads/0.5.5/dde-top-panel-0.5.5/build/frame/dde-top-panel_autogen/KH43KSYMFX/moc_dockpopupwindow.cpp:136:38: error: ‘DockPopupWindow’ has not been declared
  136 | QT_INIT_METAOBJECT const QMetaObject DockPopupWindow::staticMetaObject = { {
      |                                      ^~~~~~~~~~~~~~~
/home/zeletrik/Downloads/0.5.5/dde-top-panel-0.5.5/build/frame/dde-top-panel_autogen/KH43KSYMFX/moc_dockpopupwindow.cpp:146:20: error: ‘DockPopupWindow’ has not been declared
  146 | const QMetaObject *DockPopupWindow::metaObject() const
      |                    ^~~~~~~~~~~~~~~
/home/zeletrik/Downloads/0.5.5/dde-top-panel-0.5.5/build/frame/dde-top-panel_autogen/KH43KSYMFX/moc_dockpopupwindow.cpp:146:50: error: non-member function ‘const QMetaObject* metaObject()’ cannot have cv-qualifier
  146 | const QMetaObject *DockPopupWindow::metaObject() const
      |                                                  ^~~~~
/home/zeletrik/Downloads/0.5.5/dde-top-panel-0.5.5/build/frame/dde-top-panel_autogen/KH43KSYMFX/moc_dockpopupwindow.cpp: In function ‘const QMetaObject* metaObject()’:
/home/zeletrik/Downloads/0.5.5/dde-top-panel-0.5.5/build/frame/dde-top-panel_autogen/KH43KSYMFX/moc_dockpopupwindow.cpp:148:21: error: ‘QScopedPointer<QObjectData> QObject::d_ptr’ is protected within this context
  148 |     return QObject::d_ptr->metaObject ? QObject::d_ptr->dynamicMetaObject() : &staticMetaObject;
      |                     ^~~~~
In file included from /usr/include/x86_64-linux-gnu/qt5/QtCore/qvariant.h:51,
                 from /usr/include/x86_64-linux-gnu/qt5/QtCore/qabstractitemmodel.h:43,
                 from /usr/include/x86_64-linux-gnu/qt5/QtCore/QAbstractListModel:1,
                 from /home/zeletrik/Downloads/0.5.5/dde-top-panel-0.5.5/build/frame/dde-top-panel_autogen/LCMQWNMMZ3/../../../../appmenu/appmenumodel.h:25,
                 from /home/zeletrik/Downloads/0.5.5/dde-top-panel-0.5.5/build/frame/dde-top-panel_autogen/LCMQWNMMZ3/moc_appmenumodel.cpp:9,
                 from /home/zeletrik/Downloads/0.5.5/dde-top-panel-0.5.5/build/frame/dde-top-panel_autogen/mocs_compilation.cpp:2:
/usr/include/x86_64-linux-gnu/qt5/QtCore/qobject.h:433:33: note: declared protected here
  433 |     QScopedPointer<QObjectData> d_ptr;
      |                                 ^~~~~
In file included from /home/zeletrik/Downloads/0.5.5/dde-top-panel-0.5.5/build/frame/dde-top-panel_autogen/mocs_compilation.cpp:22:
/home/zeletrik/Downloads/0.5.5/dde-top-panel-0.5.5/build/frame/dde-top-panel_autogen/KH43KSYMFX/moc_dockpopupwindow.cpp:148:21: error: invalid use of non-static data member ‘QObject::d_ptr’
  148 |     return QObject::d_ptr->metaObject ? QObject::d_ptr->dynamicMetaObject() : &staticMetaObject;
      |                     ^~~~~
In file included from /usr/include/x86_64-linux-gnu/qt5/QtCore/qvariant.h:51,
                 from /usr/include/x86_64-linux-gnu/qt5/QtCore/qabstractitemmodel.h:43,
                 from /usr/include/x86_64-linux-gnu/qt5/QtCore/QAbstractListModel:1,
                 from /home/zeletrik/Downloads/0.5.5/dde-top-panel-0.5.5/build/frame/dde-top-panel_autogen/LCMQWNMMZ3/../../../../appmenu/appmenumodel.h:25,
                 from /home/zeletrik/Downloads/0.5.5/dde-top-panel-0.5.5/build/frame/dde-top-panel_autogen/LCMQWNMMZ3/moc_appmenumodel.cpp:9,
                 from /home/zeletrik/Downloads/0.5.5/dde-top-panel-0.5.5/build/frame/dde-top-panel_autogen/mocs_compilation.cpp:2:
/usr/include/x86_64-linux-gnu/qt5/QtCore/qobject.h:433:33: note: declared here
  433 |     QScopedPointer<QObjectData> d_ptr;
      |                                 ^~~~~
In file included from /home/zeletrik/Downloads/0.5.5/dde-top-panel-0.5.5/build/frame/dde-top-panel_autogen/mocs_compilation.cpp:22:
/home/zeletrik/Downloads/0.5.5/dde-top-panel-0.5.5/build/frame/dde-top-panel_autogen/KH43KSYMFX/moc_dockpopupwindow.cpp:148:50: error: ‘QScopedPointer<QObjectData> QObject::d_ptr’ is protected within this context
  148 |     return QObject::d_ptr->metaObject ? QObject::d_ptr->dynamicMetaObject() : &staticMetaObject;
      |                                                  ^~~~~
In file included from /usr/include/x86_64-linux-gnu/qt5/QtCore/qvariant.h:51,
                 from /usr/include/x86_64-linux-gnu/qt5/QtCore/qabstractitemmodel.h:43,
                 from /usr/include/x86_64-linux-gnu/qt5/QtCore/QAbstractListModel:1,
                 from /home/zeletrik/Downloads/0.5.5/dde-top-panel-0.5.5/build/frame/dde-top-panel_autogen/LCMQWNMMZ3/../../../../appmenu/appmenumodel.h:25,
                 from /home/zeletrik/Downloads/0.5.5/dde-top-panel-0.5.5/build/frame/dde-top-panel_autogen/LCMQWNMMZ3/moc_appmenumodel.cpp:9,
                 from /home/zeletrik/Downloads/0.5.5/dde-top-panel-0.5.5/build/frame/dde-top-panel_autogen/mocs_compilation.cpp:2:
/usr/include/x86_64-linux-gnu/qt5/QtCore/qobject.h:433:33: note: declared protected here
  433 |     QScopedPointer<QObjectData> d_ptr;
      |                                 ^~~~~
In file included from /home/zeletrik/Downloads/0.5.5/dde-top-panel-0.5.5/build/frame/dde-top-panel_autogen/mocs_compilation.cpp:22:
/home/zeletrik/Downloads/0.5.5/dde-top-panel-0.5.5/build/frame/dde-top-panel_autogen/KH43KSYMFX/moc_dockpopupwindow.cpp:148:50: error: invalid use of non-static data member ‘QObject::d_ptr’
  148 |     return QObject::d_ptr->metaObject ? QObject::d_ptr->dynamicMetaObject() : &staticMetaObject;
      |                                                  ^~~~~
In file included from /usr/include/x86_64-linux-gnu/qt5/QtCore/qvariant.h:51,
                 from /usr/include/x86_64-linux-gnu/qt5/QtCore/qabstractitemmodel.h:43,
                 from /usr/include/x86_64-linux-gnu/qt5/QtCore/QAbstractListModel:1,
                 from /home/zeletrik/Downloads/0.5.5/dde-top-panel-0.5.5/build/frame/dde-top-panel_autogen/LCMQWNMMZ3/../../../../appmenu/appmenumodel.h:25,
                 from /home/zeletrik/Downloads/0.5.5/dde-top-panel-0.5.5/build/frame/dde-top-panel_autogen/LCMQWNMMZ3/moc_appmenumodel.cpp:9,
                 from /home/zeletrik/Downloads/0.5.5/dde-top-panel-0.5.5/build/frame/dde-top-panel_autogen/mocs_compilation.cpp:2:
/usr/include/x86_64-linux-gnu/qt5/QtCore/qobject.h:433:33: note: declared here
  433 |     QScopedPointer<QObjectData> d_ptr;
      |                                 ^~~~~
In file included from /home/zeletrik/Downloads/0.5.5/dde-top-panel-0.5.5/build/frame/dde-top-panel_autogen/mocs_compilation.cpp:22:
/home/zeletrik/Downloads/0.5.5/dde-top-panel-0.5.5/build/frame/dde-top-panel_autogen/KH43KSYMFX/moc_dockpopupwindow.cpp: At global scope:
/home/zeletrik/Downloads/0.5.5/dde-top-panel-0.5.5/build/frame/dde-top-panel_autogen/KH43KSYMFX/moc_dockpopupwindow.cpp:151:7: error: ‘DockPopupWindow’ has not been declared
  151 | void *DockPopupWindow::qt_metacast(const char *_clname)
      |       ^~~~~~~~~~~~~~~
/home/zeletrik/Downloads/0.5.5/dde-top-panel-0.5.5/build/frame/dde-top-panel_autogen/KH43KSYMFX/moc_dockpopupwindow.cpp: In function ‘void* qt_metacast(const char*)’:
/home/zeletrik/Downloads/0.5.5/dde-top-panel-0.5.5/build/frame/dde-top-panel_autogen/KH43KSYMFX/moc_dockpopupwindow.cpp:155:35: error: invalid use of ‘this’ in non-member function
  155 |         return static_cast<void*>(this);
      |                                   ^~~~
/home/zeletrik/Downloads/0.5.5/dde-top-panel-0.5.5/build/frame/dde-top-panel_autogen/KH43KSYMFX/moc_dockpopupwindow.cpp:156:61: error: cannot call member function ‘virtual void* Dtk::Widget::DArrowRectangle::qt_metacast(const char*)’ without object
  156 |     return Dtk::Widget::DArrowRectangle::qt_metacast(_clname);
      |                                                             ^
/home/zeletrik/Downloads/0.5.5/dde-top-panel-0.5.5/build/frame/dde-top-panel_autogen/KH43KSYMFX/moc_dockpopupwindow.cpp: At global scope:
/home/zeletrik/Downloads/0.5.5/dde-top-panel-0.5.5/build/frame/dde-top-panel_autogen/KH43KSYMFX/moc_dockpopupwindow.cpp:159:5: error: ‘DockPopupWindow’ has not been declared
  159 | int DockPopupWindow::qt_metacall(QMetaObject::Call _c, int _id, void **_a)
      |     ^~~~~~~~~~~~~~~
/home/zeletrik/Downloads/0.5.5/dde-top-panel-0.5.5/build/frame/dde-top-panel_autogen/KH43KSYMFX/moc_dockpopupwindow.cpp: In function ‘int qt_metacall(QMetaObject::Call, int, void**)’:
/home/zeletrik/Downloads/0.5.5/dde-top-panel-0.5.5/build/frame/dde-top-panel_autogen/KH43KSYMFX/moc_dockpopupwindow.cpp:161:64: error: cannot call member function ‘virtual int Dtk::Widget::DArrowRectangle::qt_metacall(QMetaObject::Call, int, void**)’ without object
  161 |     _id = Dtk::Widget::DArrowRectangle::qt_metacall(_c, _id, _a);
      |                                                                ^
/home/zeletrik/Downloads/0.5.5/dde-top-panel-0.5.5/build/frame/dde-top-panel_autogen/KH43KSYMFX/moc_dockpopupwindow.cpp:166:32: error: invalid use of ‘this’ in non-member function
  166 |             qt_static_metacall(this, _c, _id, _a);
      |                                ^~~~
/home/zeletrik/Downloads/0.5.5/dde-top-panel-0.5.5/build/frame/dde-top-panel_autogen/KH43KSYMFX/moc_dockpopupwindow.cpp: At global scope:
/home/zeletrik/Downloads/0.5.5/dde-top-panel-0.5.5/build/frame/dde-top-panel_autogen/KH43KSYMFX/moc_dockpopupwindow.cpp:177:6: error: ‘DockPopupWindow’ has not been declared
  177 | void DockPopupWindow::accept()const
      |      ^~~~~~~~~~~~~~~
/home/zeletrik/Downloads/0.5.5/dde-top-panel-0.5.5/build/frame/dde-top-panel_autogen/KH43KSYMFX/moc_dockpopupwindow.cpp:177:31: error: non-member function ‘void accept()’ cannot have cv-qualifier
  177 | void DockPopupWindow::accept()const
      |                               ^~~~~
/home/zeletrik/Downloads/0.5.5/dde-top-panel-0.5.5/build/frame/dde-top-panel_autogen/KH43KSYMFX/moc_dockpopupwindow.cpp: In function ‘void accept()’:
/home/zeletrik/Downloads/0.5.5/dde-top-panel-0.5.5/build/frame/dde-top-panel_autogen/KH43KSYMFX/moc_dockpopupwindow.cpp:179:39: error: ‘DockPopupWindow’ does not name a type
  179 |     QMetaObject::activate(const_cast< DockPopupWindow *>(this), &staticMetaObject, 0, nullptr);
      |                                       ^~~~~~~~~~~~~~~
/home/zeletrik/Downloads/0.5.5/dde-top-panel-0.5.5/build/frame/dde-top-panel_autogen/KH43KSYMFX/moc_dockpopupwindow.cpp:179:55: error: expected ‘>’ before ‘*’ token
  179 |     QMetaObject::activate(const_cast< DockPopupWindow *>(this), &staticMetaObject, 0, nullptr);
      |                                                       ^
/home/zeletrik/Downloads/0.5.5/dde-top-panel-0.5.5/build/frame/dde-top-panel_autogen/KH43KSYMFX/moc_dockpopupwindow.cpp:179:55: error: expected ‘(’ before ‘*’ token
  179 |     QMetaObject::activate(const_cast< DockPopupWindow *>(this), &staticMetaObject, 0, nullptr);
      |                                                       ^
      |                                                       (
/home/zeletrik/Downloads/0.5.5/dde-top-panel-0.5.5/build/frame/dde-top-panel_autogen/KH43KSYMFX/moc_dockpopupwindow.cpp:179:56: error: expected primary-expression before ‘>’ token
  179 |     QMetaObject::activate(const_cast< DockPopupWindow *>(this), &staticMetaObject, 0, nullptr);
      |                                                        ^
/home/zeletrik/Downloads/0.5.5/dde-top-panel-0.5.5/build/frame/dde-top-panel_autogen/KH43KSYMFX/moc_dockpopupwindow.cpp:179:58: error: invalid use of ‘this’ in non-member function
  179 |     QMetaObject::activate(const_cast< DockPopupWindow *>(this), &staticMetaObject, 0, nullptr);
      |                                                          ^~~~
/home/zeletrik/Downloads/0.5.5/dde-top-panel-0.5.5/build/frame/dde-top-panel_autogen/KH43KSYMFX/moc_dockpopupwindow.cpp: At global scope:
/home/zeletrik/Downloads/0.5.5/dde-top-panel-0.5.5/build/frame/dde-top-panel_autogen/KH43KSYMFX/moc_dockpopupwindow.cpp:183:6: error: ‘DockPopupWindow’ has not been declared
  183 | void DockPopupWindow::unusedSignal()
      |      ^~~~~~~~~~~~~~~
/home/zeletrik/Downloads/0.5.5/dde-top-panel-0.5.5/build/frame/dde-top-panel_autogen/KH43KSYMFX/moc_dockpopupwindow.cpp: In function ‘void unusedSignal()’:
/home/zeletrik/Downloads/0.5.5/dde-top-panel-0.5.5/build/frame/dde-top-panel_autogen/KH43KSYMFX/moc_dockpopupwindow.cpp:185:27: error: invalid use of ‘this’ in non-member function
  185 |     QMetaObject::activate(this, &staticMetaObject, 1, nullptr);
      |                           ^~~~
make[2]: *** [frame/CMakeFiles/dde-top-panel.dir/build.make:97: frame/CMakeFiles/dde-top-panel.dir/dde-top-panel_autogen/mocs_compilation.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:118: frame/CMakeFiles/dde-top-panel.dir/all] Error 2
make: *** [Makefile:84: all] Error 2

```

**Comment by SeptemberHX** [comment]
As I know, UbuntuDDE 20.04 comes with DDE V15, which is not supported.
dde-top-panel only supports DDE V20, which is shipped with UbuntuDDE 20.10

**Comment by zeletrik** [comment]
Sure, my fault. Adter upgrading to 20.10 it's working just fine.
